tests/lib/test_vcs.ml

```diff
@@ -0,0 +1,18 @@
+let test_git_sanitize_tag =
+  let make_test name ~input ~expected =
+    let name = "git_sanitize_tag: " ^ name in
+    let test_fun () =
+      Alcotest.(check string)
+        name expected
+        (Dune_release.Vcs.git_sanitize_tag input)
+    in
+    (name, `Quick, test_fun)
+  in
+  [
+    make_test "empty" ~input:"" ~expected:"";
+    make_test "valid" ~input:"3.3.4" ~expected:"3.3.4";
+    make_test "tilde" ~input:"3.3.4~4.10preview1"
+      ~expected:"3.3.4_TILDE_4.10preview1";
```

Comment:
How about adding a test case for the chars that are refused only if in last position?

Reply:
done